src/ServiceControl.Plugin.Nsb6.Heartbeat/Heartbeats.cs

```diff
@@ -82,33 +85,33 @@ protected override void OnStop()
                     cancellationTokenSource.Cancel();
                 }
 
-                base.OnStop();
+                return Task.FromResult(0);
             }
 
-            void NotifyEndpointStartup(HostInformation hostInfo, DateTime startupTime)
+            void NotifyEndpointStartup(DateTime startupTime)
             {
                 // don't block here since StartupTasks are executed synchronously.
-                Task.Run(() => SendEndpointStartupMessage(hostInfo, startupTime, cancellationTokenSource.Token));
+                Task.Run(() => SendEndpointStartupMessage(startupTime, cancellationTokenSource.Token));
             }
 
-            void StartHeartbeats(HostInformation hostInfo)
+            void StartHeartbeats()
             {
                 Logger.DebugFormat("Start sending heartbeats every {0}", heartbeatInterval);
-                heartbeatTimer = new Timer(x => SendHeartbeatMessage(hostInfo), null, TimeSpan.Zero, heartbeatInterval);
+                heartbeatTimer = new Timer(x => SendHeartbeatMessage(), null, TimeSpan.Zero, heartbeatInterval);
```

Comment:
You probably need an async timer here as well. Check out the custom checks. Ping me if you need help